Easy/BestTimetoBuyandSellStock1.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * Best Time to Buy and Sell Stock
+ * Say you have an array for which the ith element is the price of a given stock on day i.
+ * If you were only permitted to complete at most one transaction (ie, buy one and sell one share of the stock), design an algorithm to find the maximum profit.
+ * Tags:  Array Dynamic Programming
+ * Similar Problems:  (M) Maximum Subarray (M) Best Time to Buy and Sell Stock II (H) Best Time to Buy and Sell Stock III (H) Best Time to Buy and Sell Stock IV (M) Best Time to Buy and Sell Stock with Cooldown
+ * O(n) time, O(1) space
+ * @author chenshuna
+ */ 
+
+public class BestTimetoBuyandSellStock {
+    public static int maxProfit(int[] prices) {
+        if(prices == null || prices.length == 0 || prices.length == 1){
+            return 0;
+        }
+        int max = 0;
+        int minPrice = prices[0];
+        for(int i = 0; i < prices.length; i++){
```

Comment:
i can start from 1 as you make sure that `prices.length > 1` and max won't be updated until 1